src/taggle.js

```diff
@@ -268,17 +270,20 @@
                 return;
             }
 
+            if (!settings.onBeforeTagAdd(e, val)) {
```

Comment:
The idea is that if something goes wrong in «onbefore» callback addition of a tag stops. 
This is why by default it returns «1» in case no callback passed at object initialization. 
If i remove this if condition the meaning of «onBefore» callback will be lost. Actually i do server validation there to check if it worth adding (marking as added) tag on a client.